frontend/src/features/admin-form/share/ShareFormModal.tsx

```diff
@@ -193,10 +193,15 @@ export const ShareFormModal = ({
       setGoLinkSaved(true)
       setGoLinkSuffixInput(goLinkSuffixData?.goLinkSuffix ?? '')
       setGoLinkHelperText(goLinkClaimSuccessHelperText)
+    } else {
```

Comment:
nit: Would probably be more intuitive to handle the reset when we close the component.

```
  const handleOnClose = () => {
    onClose()
    setGoLinkSaved(false)
    setGoLinkSuffixInput('')
    setGoLinkHelperText(undefined)
  }
  ...
    return (
    <Modal size={modalSize} isOpen={isOpen} onClose={handleOnClose}>
  ```

Reply:
Oh actually, instead of this, maybe we can add a `return` statement as cleanup for the `useEffect` hook ([doc](https://react.dev/learn/synchronizing-with-effects#step-3-add-cleanup-if-needed))! Sorry, I forgot that such a thing existed.

Reply:
@sebastianwzq Unfortunately we can't do it on modal close 🥲, as we could potentially set the admin dashboard form meta as a different form without opening the share modal, as it lies within the `WorkspaceRowsContext` -> this will cause our use effect hook to update the states.

To replicate
1. On the dashboard, on a form with existing go gov link
2. Click manage form admins (this is a different modal but this causes the useEffect hook to update the underlying state in the `ShareFormModal`)
3. Go to another form without existing go gov link and click share form
4. The Gogov link will be there

Reply:
followed @LinHuiqing's suggestion! [be769c0](https://github.com/opengovsg/FormSG/pull/6776/commits/be769c07848609aa934d5ae723756cf02a9895e8)